### PR TITLE
[Sprint 53] [1.2.x] XD-2957 Added Optimizing Serialization

### PR DESCRIPTION
**Pull request by dturanski**
Docs

### CODE REVIEW

File: src/docs/asciidoc/Optimizing-Serialization.asciidoc [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+magnitude longer. To put this in perspective, let's suppose that serialization
+and deserialization of your payload each take 5000 nanoseconds.  For a stream
+,e.g., `some-source | some-sink` running in a distributed XD runtime, moving the
+payload from the source to the sink requires serialization at the producer and
+deserialization at the consumer. Both operations would account for 10,000 ns or
+10 microseconds. So the maximum theoretical throughput for this simple stream,
+with no partitioning or parallel deployment, is at most 100,000 messages/sec.
+This does not account for network latency (transporting the serialized bytes),
+overhead in the messaging middleware, or computations performed by the modules.
+If your application requires this level of throughput then serialization is a
```
Comment (cppwfs) [inline src/docs/asciidoc/Optimizing-Serialization.asciidoc]: Slight rewording.
If your application requires a higher level of throughput than mentioned above then serialization is a potential bottleneck.

File: src/docs/asciidoc/Optimizing-Serialization.asciidoc [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+demanding, serialization performance becomes less of an issue. In this
+hypothetical scenario, At a throughput of 10,000 messages/second, 10
+microseconds is only 10% of the maximum time required to process the message.
+However if you trying to stream large objects at this rate, serialization time,
+along with transport and processing time, will be significantly higher.
+
+=== Serialization in XD
+
+NOTE: This content applies to Spring XD 1.2.x. In Spring XD 2.0, these
+components work in a similar way but will be packaged and configured differently.
```
Comment (cppwfs) [inline src/docs/asciidoc/Optimizing-Serialization.asciidoc]: Fielders choice... Not sure the XD 2.0 comment belongs in the 1.2.x docs.

**Comment by dturanski** [comment]
Both excellent points

File: src/docs/asciidoc/Optimizing-Serialization.asciidoc [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+The `Serializer` interface exposes `Kryo`, `Input`, and `Output` which provide
+complete control over which fields are included and other internal settings as
+described in the https://github.com/EsotericSoftware/kryo[documentation].
+
+
+==== Disabling References
+
+A simple setting may be applied to boost performance if your payload types do
+not contain any cyclical references. Kryo needs to do some extra work in this
```
Comment (cppwfs) [inline src/docs/asciidoc/Optimizing-Serialization.asciidoc]: Reworded to make this clearer.
If payload contains cyclical references Kryo needs to do some extra work in this case and the https://github.com/EsotericSoftware/kryo#references[references]
+setting is enabled by default.

**Comment by cppwfs** [comment]
LGTM I'll update the doc with the comments upon merge.

**Comment by cppwfs** [comment]
Merged

**Pull request closed without merging**